src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp

```diff
@@ -21,121 +21,133 @@ namespace opossum {
 
 class PredicatePlacementRuleTest : public StrategyBaseTest {
  protected:
+  static void SetUpTestSuite() {
+    _table_a = load_table("resources/test_data/tbl/int_float.tbl");
+    _table_b = load_table("resources/test_data/tbl/int_float2.tbl");
+    _table_c = load_table("resources/test_data/tbl/int_float3.tbl");
+    _table_d = load_table("resources/test_data/tbl/int_int3.tbl");
+    _table_e = load_table("resources/test_data/tbl/int_int4.tbl");
+  }
+
   void SetUp() override {
-    Hyrise::get().storage_manager.add_table("a", load_table("resources/test_data/tbl/int_float.tbl"));
-    _table_a = std::make_shared<StoredTableNode>("a");
-    _a_a = _table_a->get_column("a");
-    _a_b = _table_a->get_column("b");
-
-    Hyrise::get().storage_manager.add_table("b", load_table("resources/test_data/tbl/int_float2.tbl"));
-    _table_b = std::make_shared<StoredTableNode>("b");
-    _b_a = _table_b->get_column("a");
-    _b_b = _table_b->get_column("b");
-
-    Hyrise::get().storage_manager.add_table("c", load_table("resources/test_data/tbl/int_float3.tbl"));
-    _table_c = std::make_shared<StoredTableNode>("c");
-    _c_a = _table_c->get_column("a");
-    _c_b = _table_c->get_column("b");
-
-    Hyrise::get().storage_manager.add_table("d", load_table("resources/test_data/tbl/int_int3.tbl"));
-    _table_d = std::make_shared<StoredTableNode>("d");
-    _d_a = _table_d->get_column("a");
-    _d_b = _table_d->get_column("b");
-
-    Hyrise::get().storage_manager.add_table("e", load_table("resources/test_data/tbl/int_int4.tbl"));
-    _table_e = std::make_shared<StoredTableNode>("e");
-    _e_a = _table_e->get_column("a");
-    _e_b = _table_e->get_column("b");
+    Hyrise::get().storage_manager.add_table("a", _table_a);
+    _stored_table_a = StoredTableNode::make("a");
+    _a_a = _stored_table_a->get_column("a");
+    _a_b = _stored_table_a->get_column("b");
```

Comment:
Now that I am seeing this again, I realize that the proper way to work around the reset of the StorageManager would have been to either use a MockNode with statistics or a StaticTableNode. No need to change this again, just wanted to let you know.